### PR TITLE
Adform adapter passes gdpr and digitrust to bid request

### PR DESCRIPTION
**Pull request by volhalink**

### CODE REVIEW

File: adapters/adform/adform.go [hunk 1]
```diff
@@ -219,12 +262,33 @@ func (r *adformRequest) buildAdformUrl(a *AdformAdapter) string {
 	if r.isSecure {
 		uri = strings.Replace(uri, "http://", "https://", 1)
 	}
-	adid := ""
+	var Url *url.URL
```
Comment (vstatkevich) [inline adapters/adform/adform.go]: need the second step of refactoring to increase readability (e.g. `uri`, `Url`, try add `adUnitParams` to `parameters`).

File: adapters/adform/adform.go [hunk 2]
```diff
@@ -285,7 +359,7 @@ func NewAdformBidder(client *http.Client, endpointURL string) *AdformAdapter {
 	return &AdformAdapter{
 		http:    a,
 		URI:     endpointURL,
-		version: "0.1.1",
+		version: "0.1.2",
```
Comment (vstatkevich) [inline adapters/adform/adform.go]: 👍

File: adapters/adform/adform.go [hunk 3]
```diff
@@ -367,6 +441,40 @@ func openRtbToAdformRequest(request *openrtb.BidRequest) (*adformRequest, []erro
...
 
+	gdprApplies := ""
+	var extRegs openrtb_ext.ExtRegs
+	if request.Regs != nil {
+		if err := json.Unmarshal(request.Regs.Ext, &extRegs); err != nil {
+			errors = append(errors, &adapters.BadInputError{
+				Message: err.Error(),
+			})
+		}
+		if extRegs.GDPR != nil && (*extRegs.GDPR == 0 || *extRegs.GDPR == 1) {
```
Comment (vstatkevich) [inline adapters/adform/adform.go:455]: Looks like we already have this check in `auction.go` `validateRegs`

File: pbs/pbsrequest.go [hunk 4]
```diff
@@ -378,7 +379,16 @@ func (req *PBSRequest) ParseGDPR() string {
 	if req == nil || req.Regs == nil {
 		return ""
 	}
-	return parseString(req.Regs.Ext, "gdpr")
+	if len(req.Regs.Ext) == 0 {
```
Comment (vstatkevich) [inline pbs/pbsrequest.go]: move to `if req == nil || req.Regs == nil {`

File: adapters/adform/adform.go (hunk 1 above)
Comment (volhalink) [reply adapters/adform/adform.go]: in `net/url`'s `URL` object query string parameters are key-value pairs and `adUnitParams` in query string are just values as we convert the pair to base64 value. So we can't use the same approach as for regular parameters

**Pull request closed without merging**